### PR TITLE
Add RemoteAddr in /AuthZPlugin.AuthZReq

### PR DESCRIPTION
**Pull request by ghost**
**- What I did**
Add "RemoteAddr" field to authorization Request struct. That will allow to get client IP address with authorization plugin and could be used for logging purposes.

### CODE REVIEW

**Comment by juliengk** [comment]
any comment on that?

**Comment by tonistiigi** [comment]
LGTM

**Comment by thaJeztah** [comment]
Discussing this in the maintainers meeting, and while we don't have a problem adding this, we were wondering if you tried using this information; there's many cases where the remote address may not be the actual remote address (e.g., proxy being used, or classic swarm, or the unix-socket being used)

Perhaps we should document that limitation as part of this PR

Moving to code review

**Comment by juliengk** [comment]
what about checking for X-Forwarded-For header when proxy is being used?

remoteAddr := r.RemoteAddr
if ip := r.Header.Get("X-Forwarded-For"); ip != "" {
                remoteAddr = ip
}

authCtx := NewCtx(plugins, user, userAuthNMethod, r.Method, r.RequestURI, remoteAddr)

**Comment by thaJeztah** [comment]
I'm ok with adding that (but still make sure we document the limitations properly); also document that this information can be "spoofed" (which may be important information, given that it's used for Authz)

**Comment by juliengk** [comment]
Is it good like this with comment? Or do you want to have it mentionned somewhere else?

File: pkg/authorization/middleware.go [hunk 1]
```diff
@@ -71,7 +71,12 @@ func (m *Middleware) WrapHandler(handler func(ctx context.Context, w http.Respon
 			userAuthNMethod = "TLS"
 		}
 
-		authCtx := NewCtx(plugins, user, userAuthNMethod, r.Method, r.RequestURI)
+		remoteAddr := r.RemoteAddr
+		if ip := r.Header.Get("X-Forwarded-For"); ip != "" {
```
Comment (stevvooe) [inline pkg/authorization/middleware.go]: This is more complicated that this: you can handle this but also `Forwarded`. ;)

File: pkg/authorization/api.go [hunk 2]
```diff
@@ -65,6 +65,11 @@ type Request struct {
 	// RequestPeerCertificates stores the request's TLS peer certificates in PEM format
 	RequestPeerCertificates []*PeerCertificate `json:"RequestPeerCertificates,omitempty"`
 
+	// RemoteAddr stores the network address that sent the request
+	// There's many cases where the remote address may not be the actual
+	// remote address (e.g., proxy being used, or classic swarm, or the unix-socket being used)
+	RemoteAddr string `json:"RemoteAddr,omitempty"`
```
Comment (thaJeztah) [inline pkg/authorization/api.go]: Discussing with @anusha-ragunathan - perhaps this should be called `RequestAddr` (i.e., where the request comes from) ?

File: pkg/authorization/middleware.go (hunk 1 above)
Comment (thaJeztah) [reply pkg/authorization/middleware.go]: See https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Forwarded for how those headers look; looks like we need some additional parsing for that

File: pkg/authorization/api.go (hunk 2 above)
Comment (juliengk) [reply pkg/authorization/api.go]: field name has been changed.

**Comment by olljanat** [comment]
Derek add label: rebuild/*

**Comment by olljanat** [comment]
@juliengk CI build fails to error:
```
18:49:14 pkg/authorization/authz.go:78:3:warning: unknown field RemoteAddr in struct literal (gosimple)
18:49:14 pkg/authorization/authz.go:78:3:warning: unknown field RemoteAddr in struct literal (interfacer)
18:49:14 pkg/authorization/authz.go:78:3:warning: unknown field RemoteAddr in struct literal (unconvert)
18:49:14 Build step 'Execute shell' marked build as failure
```
plz check

**Pull request closed without merging**